README.md

```diff
@@ -30,8 +30,7 @@ your answer to this markdown file.
 
 ## Answer
 
-Something that is always true at the start of each recursive call is the the Fibonacci numbers for n = 0 and n = 1 are correct. Since we know that this is true. We can see that at the start of the recursive call fib(n - 1). The array already 
-contains the Fibbonacci numbers of n = 0 and n = 1. This ensures that we can compute the next Fibbonacci number by using the the values for arr[n - 1] and arr[n - 2]. We can now add them to get fib(n). 
+Something that is always true at the start of each recursive call is the the Fibonacci numbers for $n = 0$ and $n = 1$ are correct. When we make recursive calls to fib(n - 1). The calls are made till we reach our base case. Then the array is progressively built up to $n - 1$. This ensures that we can compute the next Fibbonacci number $fib(n)$, by using the the values for $arr[n - 1]$ and $arr[n - 2]$, Which are the values of $fib(n - 1)$ and $fib(n - 2)$. We can now add them to get $fib(n)$. 
```

Comment:
What has to be true before you run your code to compute the next Fibonacci number (line 8 in your code) for that computation to be correct?

Reply:
We need to have the array built off of the base case. Are you referring to the recursion?

Reply:
You can't use the base case for an invariant. You need to say something about the current iteration, not the first few iterations.

Reply:
Oh, ok I think I get it. So the invariant should something like that 
arr[1 . . . n - 1] are already computed.